src/Universum/Exception.hs

```diff
@@ -19,6 +24,25 @@ import Control.Applicative (Applicative (pure))
 import Control.Monad.Except (MonadError, throwError)
 import Data.Maybe (Maybe, maybe)
 
+#if ( __GLASGOW_HASKELL__ >= 800 )
+import GHC.Show (Show)
+import GHC.Stack (CallStack, HasCallStack)
+
+import qualified Control.Exception.Safe as Safe (impureThrow, toException)
+
+-- | Type that represents exceptions used in cases when a particular codepath
+-- is not meant to be ever executed, but happens to be executed anyway.
+data Bug = Bug SomeException CallStack
+    deriving (Show)
+
+instance Exception Bug
```

Comment:
Let's have a proper `displayException` implementation here. We should use the `prettyCallStack` function from `GHC.Stack` to render the call stack, `displayException` to render the stored `SomeException`, and concatenate the two using with `"\n"` in between (`SomeException` comes first, then `"\n"`, then pretty-printed `CallStack`).

This would match the behavior of `error`.